runtime/jvmti/jvmtiThread.c

```diff
@@ -466,11 +466,17 @@ jvmtiGetThreadInfo(jvmtiEnv* env,
 			jobject contextClassLoader;
 #if JAVA_SPEC_VERSION >= 19
 			j9object_t threadHolder = J9VMJAVALANGTHREAD_HOLDER(currentThread, threadObject);
```

Comment:
this can be fetched later, once we know we are not in the `isVirtual` path

Reply:
Do you mean at line 558? If `J9_ARE_ALL_BITS_SET(vm->jclFlags, J9_JCL_FLAG_THREADGROUPS)` is not taken then it will not be fetched. threadHolder is also needed in 2 later locations.
Unless you want
```c
j9object_t threadHolder = NULL;
BOOLEAN isVirtual = IS_VIRTUAL_THREAD(currentThread, threadObject);
if (!isVirtual) {
    threadHolder = J9VMJAVALANGTHREAD_HOLDER(currentThread, threadObject);
}
```

Reply:
okay, I didnt realize its used outside 558, I think its fine as is